src/DIRAC/WorkloadManagementSystem/Service/TornadoPilotLoggingHandler.py

```diff
@@ -0,0 +1,109 @@
+""" Tornado-based HTTPs JobMonitoring service.
+"""
+
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+__RCSID__ = "$Id$"
+
+import os, json
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Tornado.Server.TornadoService import TornadoService
+from DIRAC.Core.DISET.RequestHandler import RequestHandler, getServiceOption
+from DIRAC.Core.Utilities.ObjectLoader import ObjectLoader
+
+sLog = gLogger.getSubLogger(__name__)
+
+
+class TornadoPilotLoggingHandler(TornadoService):
+    log = sLog
+
+    @classmethod
+    def initializeHandler(cls, infoDict):
+        """
+        Called once, at the first request. Create a directory where pilot logs will be stored.
+
+        :param infoDict:
+        :return: None
+        """
+
+        cls.log.info("Handler initialised ...")
+        cls.log.debug("with a dict: ", str(infoDict))
+        defaultOption, defaultClass = "LoggingPlugin", "BasicPilotLoggingPlugin"
+        configValue = getServiceOption(infoDict, defaultOption, defaultClass)
+
+        result = ObjectLoader().loadObject("WorkloadManagementSystem.Service.%s" % (configValue,), configValue)
+        if not result["OK"]:
+            cls.log.error("Failed to load LoggingPlugin", "%s: %s" % (configValue, result["Message"]))
+            return result
+
+        componentClass = result["Value"]
+        cls.loggingPlugin = componentClass()
+        cls.log.info("Loaded: PilotLoggingPlugin class", configValue)
+
+        cls.meta = {}
+        logPath = os.path.join(os.getcwd(), "pilotlogs")
+        cls.meta["LogPath"] = logPath
+        if not os.path.exists(logPath):
+            os.makedirs(logPath)
+        cls.log.info("Pilot logging directory:", logPath)
+
+    def initializeRequest(self):
+        """
+        Called for each request.
+
+        :return: None
+        """
+
+        self.log.info("Request initialised.. ")
+
+    auth_sendMessage = ["Operator", "Pilot", "GenericPilot"]
```

Comment:
I would put all the `auth_*` in the ConfigTemplate.cfg, not in the code.

Reply:
OK, I prepared this. The question is, if those restrictions should be removed from the code (according to the docs, they would be ignored if present in the configuration) or should they be kept as a bare minimum ?

Reply:
Honestly I would remove them from the code.

Reply:
I've done it this afternoon.